runtime/src/main/java/org/corfudb/runtime/Backup.java

```diff
@@ -265,11 +290,36 @@ private void addToTarFile(File tableFile,TarArchiveOutputStream tarOutput) throw
                 .map(tableName -> CorfuRuntime
                         .getStreamID(TableRegistry.getFullyQualifiedTableName(tableName)))
                 .collect(Collectors.toList());
-        log.info("{} tables need to be backed up: {}", tables.size(), tables);
+        log.info("{} tables need to be backed up.", tables.size());
 
         return tables;
     }
 
+    /**
+     * Get UUIDs of all registered UFO tables
+     *
+     * @return  List<UUID>   - a list of UUIDs for all registered UFO tables
+     */
+    private List<UUID> getAllTablesUUID() {
+        return runtime.getTableRegistry().listTables().stream()
+                .map(table -> CorfuRuntime.getStreamID(
+                        TableRegistry.getFullyQualifiedTableName(table.getNamespace(), table.getTableName())))
```

Comment:
can we add `getFullyQualifiedTableName` method into `table`? then it will be much cleaner: `table.getFullyQualifiedTableName()` and more logically consistent

Reply:
You are right that it can make code looks cleaner. 
`listTables()` returns a collection of TableNames (protobuf message) whose only fields are `namespace` and `table_name`, and `getFullyQualifiedTableName()` is application logic that determines how we present the full name. If we add a new field `fully_qualified_table_name` in `TableName` that would introduce data duplication and stronger coupling of data and logic.

Reply:
there is no coupling. Actually, the current code creates more boilerplate code than it should be. 
The table always knows its namespace and its own name, so do you really believe `$` symbol makes code more fragile?
Actually, it's vise versa - in the table class we define a contact - what is a full name of the table. Otherwise how a developer should guess that there is a Utility class somewhere, which keeps that information about the contract for the naming?

![image](https://user-images.githubusercontent.com/314292/116168548-82e1e380-a6b7-11eb-854f-cb0502adfbb7.png)

Reply:
Just to make sure that we are on the same page -- if we want to do `table.getFullyQualifiedTableName()`, we have to add a field in TableName (table is a protobuf message).
```
// Table name.
message TableName {
    string namespace = 1;
    string table_name = 2;
    string fully_qualified_name = 3;
}
```
If one day we need to change the connector from `$` to `&`, or add support for another kind of tables besides UFO tables which uses whichever other symbol, we will have to do DB migration and rewrite the `fully_qualified_name` field. In other words, we are hardcoding the rule/logic of generating fully_qualified_name (`TableRegistry. getFullyQualifiedTableName`) into actual DB tables.

Reply:
oh, ok, didn't know that Table is a protobuf class